File: Python/7py 1/contain_all_rots.py
"""
|--------------------------------------------------------------------------
| All Inclusive?
|--------------------------------------------------------------------------
|
"""

def contain_all_rots(strng, arr):
   count = 0
   for i in range(len(strng)):
      if strng[i:len(strng)]+strng[0:i] in arr: count += 1
   return count == len(strng)


def contain_all_rots(s, l):
   return all(s[i:]+s[:i] in l for i in range(len(s)))


def contain_all_rots(strng, arr):
   if strng == "": return True
   i, l = 0, len(strng)
   while i < l:
      strng = strng[-1]+strng[:-1]
      if not strng in arr: return False
      i += 1
   return True


print(contain_all_rots("", []), True)
print(contain_all_rots("", ["bsjq", "qbsj"]), True)
print(contain_all_rots("bsjq", ["bsjq", "qbsj", "sjqb", "twZNsslC", "jqbs"]), True)
print(contain_all_rots("XjYABhR", ["TzYxlgfnhf", "yqVAuoLjMLy", "BhRXjYA", "YABhRXj", "hRXjYAB", "jYABhRX", "XjYABhR", "ABhRXjY"]), False)

File: JavaScript/7js 1/containAllRots.js
/*
|--------------------------------------------------------------------------
| All Inclusive?
|--------------------------------------------------------------------------
|
*/

function containAllRots(strng, arr) {
   let count = 0;
   for (let i = 0; i < strng.length; i++) {
      x = strng[i] + strng.slice(i + 1, strng.length) + strng.slice(0, i);
      if (arr.includes(x)) count += 1;
   }
   return count == strng.length;
}


function containAllRots(strng, arr) {
   return [...strng]
      .map((_, i) => strng.substr(i) + strng.slice(0, i))
      .every((x) => arr.includes(x));
}


function containAllRots(str, arr) {
   for (let i = 0; i < str.length; i++) {
      if (arr.indexOf(str.slice(i) + str.slice(0, i)) === -1) {
         return false;
      }
   }
   return true;
}


console.log(containAllRots("", []), true);
console.log(containAllRots("", ["bsjq", "qbsj"]), true);
console.log(containAllRots("bsjq", ["bsjq", "qbsj", "sjqb", "twZNsslC", "jqbs"]), true);
console.log(
   containAllRots("XjYABhR", [
      "TzYxlgfnhf",
      "yqVAuoLjMLy",
      "BhRXjYA",
      "YABhRXj",
      "hRXjYAB",
      "jYABhRX",
      "XjYABhR",
      "ABhRXjY",
   ]),
   false
);